eloquentjs/4_4_deep_comparison.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Performs deep comparison of two inputs and determines equality
+ * @param {*} itemA 
+ * @param {*} itemB 
+ * @returns 
+ */
+function deepEqual(itemA, itemB) {
+    if(itemA === itemB) {
+        return true;
+    }
+    if (typeof itemA !== typeof itemB) {
+        return false;
+    }
+    if(itemA === null || itemB === null) {
+        return false;
+    }
+    if (typeof itemA !== 'object') {
+        return false;
+    }
+
+    let entriesA = Object.entries(itemA);
+    let entriesB = Object.entries(itemB);
+
+    if (entriesA.length !== entriesB.length) {
+        return false;
+    }
+
+    entriesA.sort();
```

Comment:
This is a good enough solution for now, but you can actually do it in linear time by checking to make sure they both have the same number of keys and then iterating overt the keys in one and using constant time access into the other object to make sure they have equal values.

Also, just for your understanding, sort may work in a way you're not expecting: https://replit.com/@SKLARGBLAR/objkeysexampe#index.js